packages/edit-site/src/components/page-patterns/style.scss

```diff
@@ -98,6 +98,11 @@
 
 		.edit-site-patterns__title {
 			min-height: $grid-unit-50;
+
+			.components-heading {
+				flex-grow: 1;
+				flex-basis: 0;
+			}
```

Comment:
Without this style, the add button would be pushed out.

![image](https://github.com/WordPress/gutenberg/assets/54422211/5be6dc8c-9b79-4cb8-9cda-dfb7de7fab6e)